src/main/java/bio/terra/service/upgrade/flight/UpgradeProfileFlight.java

```diff
@@ -0,0 +1,27 @@
+package bio.terra.service.upgrade.flight;
+
+import bio.terra.model.UpgradeModel;
+import bio.terra.service.iam.AuthenticatedUserRequest;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.profile.ProfileService;
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import org.springframework.context.ApplicationContext;
+
+public class UpgradeProfileFlight extends Flight {
+
+    public UpgradeProfileFlight(FlightMap inputParameters, Object applicationContext) {
+        super(inputParameters, applicationContext);
+
+        ApplicationContext appContext = (ApplicationContext) applicationContext;
+        ProfileService profileService = (ProfileService) appContext.getBean("profileService");
```

Comment:
Is this the usual convention for getting a bean by name in jade code? One way to clean this up a little (and to avoid code duplication) is to make this a static function on `ProfileService` such as `ProfileService.fromContext()`.

Reply:
This is the usual (ugly) convention. I like the idea of your way to tidy that up.

Stairway is Spring-agnostic, so it just passes through an Object so that cast has to be somewhere. Putting into statics inside of the classes would be an improvement.